soundbay/utils/logging.py

```diff
@@ -125,11 +128,43 @@ def calc_metrics(self, epoch: int, mode: str = 'train', label_names: List[str] =
             wandb.log({f'{mode}_charts/conf_mat': wandb.plot.confusion_matrix(probs=None, y_true=self.label_list,
                                                                               preds=self.pred_list,
                                                                               class_names=label_names)},
-                                                                              step=epoch)
+                                                                              step=epoch, commit=True)
         self.pred_list = []  # flush
         self.label_list = []
         self.pred_proba_list = []
 
+    def upload_artifacts(self, audio: torch.Tensor, label: torch.Tensor, raw_wav: torch.Tensor, idx: torch.Tensor, sample_rate: int=16000, flag: str='train', epoch: int=1):
+        """upload algorithm artifacts to W&B during training session"""
+        matplotlib.use('Agg')
+        idx = idx.detach().cpu().numpy()
+        label = label.detach().cpu().numpy()
+
+
+        # Original wavs batch
+
+        artifact_wav = torch.squeeze(raw_wav).detach().cpu().numpy() * 100
+        artifact_wav_list = [artifact_wav[x,...] for x in range(artifact_wav.shape[0])]
```

Comment:
I think you can directly iterate on the numpy object, no need to create a list for that